sdks/how-to/how-to-create-a-user-from-a-patient.md

```diff
@@ -0,0 +1,151 @@
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:instantiate a message factory-->
+```typescript
+const messageFactory = new ICureRegistrationEmail(
+  hcp,
+  'URL_WHERE_TO_LOGIN',
+  'SOLUTION_NAME',
+  patient,
+)
+```
+
+Then, it is possible to invite the User using the newly created factory object and the existing Patient.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:doctor invites user-->
+```typescript
+await api.userApi.createAndInviteUser(patient, messageFactory, 3600)
+```
+
+### The User Logs in and Asks for Access
+
+After that, the User will receive an email or a SMS message that contains their login and a short-lived authentication 
+token that they can use to log in.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:user logs in-->
+```typescript
+const anonymousMedTechApi = await new AnonymousMedTechApiBuilder()
+  .withICureBaseUrl(host)
+  .withMsgGwUrl(msgGtwUrl)
+  .withMsgGwSpecId(specId)
+  .withCrypto(webcrypto as any)
+  .withAuthProcessByEmailId(authProcessId)
+  .withAuthProcessBySmsId(authProcessId)
+  .build()
+
+const { publicKey, privateKey } = await anonymousMedTechApi.cryptoApi.RSA.generateKeyPair()
+const publicKeyHex = ua2hex(await anonymousMedTechApi.cryptoApi.RSA.exportKey(publicKey, 'spki'))
+const privateKeyHex = ua2hex(await anonymousMedTechApi.cryptoApi.RSA.exportKey(privateKey, 'pkcs8'))
+
+await anonymousMedTechApi.authenticationApi.authenticateAndAskAccessToItsExistingData(
+  patientUsername,
+  patientToken,
+  async () => ({ privateKey: privateKeyHex, publicKey: publicKeyHex }),
+)
+```
+
+The `authenticateAndAskAccessToItsExistingData` method will set up the User private and public key, it will create a 
+long-lived authentication token, and will send a Notification to all the Healthcare Professionals that have a delegation 
+for the Patient to ask access to their data.
+
+### The Doctor Receives the Notification and Gives Back Access
+
+Once the Notification is sent, on the Doctor side you can use the MedTech SDK to filter his Notifications and get the one related to 
+the new User.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:doctor gets pending notifications-->
+```typescript
+const newNotifications = await api.notificationApi.getPendingNotifications()
+const patientNotification = newNotifications.filter(
+  (notification) =>
+    notification.type === NotificationTypeEnum.NEW_USER_OWN_DATA_ACCESS &&
+    notification.responsible === patient.id,
+)[0]
+```
+
+Then, you can change the status of the Notification to signal that the operation is being taken care of.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:notification set ongoing-->
+```typescript
+const ongoingStatusUpdate = await api.notificationApi.updateNotificationStatus(
+  patientNotification,
+  'ongoing',
+)
+```
```

Comment:
Is it necessary to update the notification status to ongoing? What happens if I don't?

Reply:
It is not strictly necessary, but it helps categorizing the notifications